js/script.js

```diff
@@ -110,76 +110,69 @@ const determineShareUrl = (windowUrl, jsonCityName) => {
-            const popup = L.popup({
-              pane: "fixed",
-              className: "popup-fixed",
-              autoPan: false,
-            }).setContent(popupContent);
-            layer.bindPopup(popup).openPopup();
-            layer.bindPopup(popupContent).openPopup();
-          },
-        });
-      },
-    }).addTo(map);
-
-    citiesArray.sort((a, b) => a.localeCompare(b));
-    citiesArray.forEach((city) =>
-      $("#city-choice").append(
-        $("<option></option>").attr("value", city).text(city)
-      )
-    );
+/**
+ * Move the map to the city boundaries and set its score card.
+ *
+ * @param map: The Leaflet map instance.
+ * @param city: An object with a `layout` key (Leaflet value) and keys representing the
+ *    score card properties stored in the Geojson files.
+ */
+const setMapToCity = (map, city) => {
+  const { layer } = city;
+  map.fitBounds(layer.getBounds());
+  const popupContent = generatePopupContent(map, city);
+  const popup = L.popup({
+    pane: "fixed",
+    className: "popup-fixed",
+    autoPan: false,
+  }).setContent(popupContent);
+  layer.bindPopup(popup).openPopup();
+};
 
-    if (window.location.href.indexOf("#parking-reform-map") === -1) {
-      map.fitBounds(citiesPolygons.getBounds());
-    }
-  }).then(() => {
-    // Select default city
-    $("#city-choice").val("Columbus, OH").change();
-  });
+/**
+ * Load the cities from GeoJson and set up an event listener to change cities when the user
+ * toggles the city selection.
+ *
+ * @param map: The Leaflet map instance.
+ * @param string initialCity: the lower-case city name to initially load.
+ */
+const setUpCitiesLayer = (map, initialCity) => {
+  fetch("./data/cities-polygons.geojson")
+    .then((response) => response.json())
+    // Add the GeoJSON to the map & store the parsed data.
+    .then((data) => {
+      const cities = {};
+      L.geoJson(data, {
+        style() {
+          return citiesPolygonsStyle;
+        },
+        onEachFeature(feature, layer) {
+          const key = parseCityName(feature.properties.Name);
+          cities[key] = { layer, ...feature.properties };
+        },
+      }).addTo(map);
+      return cities;
+    })
+    .then((cities) => {
+      // Set map to update when city selection changes.
+      const cityToggleElement = document.getElementById("city-choice");
+      cityToggleElement.addEventListener("change", () => {
+        setMapToCity(map, cities[cityToggleElement.value]);
+      });
+
+      // Add each city to the city selection toggle and set the initial city.
+      const cityKeys = Object.keys(cities).sort();
+      cityKeys.forEach((key) => {
+        const option = document.createElement("option");
+        option.value = key;
+        option.textContent = cities[key].Name;
+        cityToggleElement.appendChild(option);
+      });
+
+      // Set initial city.
+      cityToggleElement.value = initialCity;
+      setMapToCity(map, cities[initialCity]);
```

Comment:
We could call this earlier, above the "Add each city to the city selection toggle" section. The benefit is that the map will more quickly render the city & its score card.

But we'll render the map without the city toggle selection being rendered, which is maybe weird?

What do you think is better behavior?